GCTC-NTGC/gc-digital-talent
# Bug - 2693 - Allow Users to Delete Skills

Resolves #2693 

## Summary

This updates the `<ExperienceForm>` component to use the `useFieldArray` hook form `react-hook-form` to handle the skills array. Previously, we were manually updating using `setValue` which was causing problems when attempting to remove items.

## Testing

1. Navigate to the frontend directory, `cd frontend`
2. Build the talent search workspace `npm run production --workspace=talentsearch`
3. Navigate to the [Skills and Experiences](http://localhost:8000/en/talent/profile/skills-and-experiences) page.
4. Edit an existing experience
5. Test adding and removing skills
  - Add with the **Add Skills** section
  - Remove skills from
    - Chips under **Skills** attached to this experience
    - List under **Add Skills** section
    - Using the **Remove from experience** button in **Skills in Detail** section

All of the listed ways to remove a skill should remove it from the interface and send the proper mutation variables to execute the sync.

## Note

While working on this PR, I noticed that the `useForm` hook should be avoided when already within a form context. If you are already within that context, we should be using `useFormContext` in-place of that to get the existing form context. `useForm` seems to create a new context  which can cause problems.
